### PR TITLE
look for python in more places on windows

### LINKED ISSUE #189: platform/bundledpython/python.exe may not exist in Cloud SDK on Windows
It then fails to run a local server.

### PR DESCRIPTION
**Pull request by elharo**
@meltsufin close #189

### CODE REVIEW

**Comment by meltsufin** [comment]
:ship:

**Comment by meltsufin** [comment]
I guess you wanted to do more than just update the description. :)
Let me know when it's ready for review.

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java [hunk 1]
```diff
@@ -196,8 +196,23 @@ public Path getJavaAppEngineSdkPath() {
...
 
-  private Path getWindowsPythonPath() {
-    return getSdkPath().resolve(WINDOWS_BUNDLED_PYTHON);
+  // https://github.com/GoogleCloudPlatform/appengine-plugins-core/issues/189
+  @VisibleForTesting
+  Path getWindowsPythonPath() {
+    String cloudSdkPython = System.getenv("CLOUDSDK_PYTHON");
+    if (cloudSdkPython != null) {
+      Path cloudSdkPythonPath = Paths.get(cloudSdkPython);
+      if (Files.isExecutable(cloudSdkPythonPath)) {
```
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java]: I think I would skip the isExecutable check. If the user tried to specify the wrong path, let it crash so they can fix their mistake.

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java [hunk 2]
```diff
@@ -196,8 +196,23 @@ public Path getJavaAppEngineSdkPath() {
...
+        return cloudSdkPythonPath;
+      }
+    }
+    
+    Path pythonPath = getSdkPath().resolve(WINDOWS_BUNDLED_PYTHON);
+    if (Files.isExecutable(pythonPath)) {
```
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java]: ditto... if it's not executable, something is really wrong with the installation of the Cloud SDK. Let it crash.

Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java]: Here, actually, a file exists check is needed.

Comment (elharo) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java]: Files.isExecutable implies exists:

https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#isExecutable(java.nio.file.Path)

Let me think about what should happen is a bad path is specified.

Comment (elharo) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java]: Under what circumstances might the windows bundled python no be found?

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java (hunk 1 above)
Comment (elharo) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java]: changed to exists

**Comment by elharo** [comment]
PTAL @meltsufin

File: src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java (hunk 2 above)
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java]: If the user explicitly chooses not to install it as part of the Windows install wizard.

**Comment by meltsufin** [comment]
LGTM

**Pull request merged**